### PR TITLE
Add capability to maximize posts

### PR DESCRIPTION
**Pull request by TJScalzo**
The main idea behind this is to add a way to find a post's url. That way, people can share the website the post talks about, and also share the fact that they discovered it through inspiring.online.

### CODE REVIEW

**Comment by TJScalzo** [comment]
Here's what it looks like:
screen shot 2016-09-28 at 6 59 47 pm

File: js/inspiring-online.js [hunk 1]
```diff
@@ -182,6 +182,14 @@ function renderPost(postData) {
...
..."text/css">.maximize-post{float:right; width:25px; opacity:.5; margin-bottom:-4px;}.maximize-post:hover{opacity:.75; width:27px; margin:-1px -1px -5px auto; transition:opacity .15s linear;}</style><g><path d="M7,54.3l15.2-15l23.3,23.5c1.2,1.2,3.1,1.2,4.2,0l13.4-13.4c1.2-1.2,1.2-3.1,0-4.3L39.2,22.4L53.7,8.2   c1.7-1.6,0.5-4.5-1.8-4.5H5.8C4,3.7,2.5,5.2,2.5,7v45.5C2.5,54.8,5.3,55.9,7,54.3z"></path><path d="M144.2,3.7H98.1c-2.3,0-3.5,2.8-1.8,4.5l14.5,14.3l-24,23c-1.2,1.2-1.2,3.1,0,4.3L100,63.1   c1.2,1.2,3.1,1.2,4.2,0l23.6-23.8l15.2,15c1.7,1.6,4.5,0.5,4.5-1.9V7C147.5,5.2,146,3.7,144.2,3.7z"></path><path d="M50.1,85.8c-1.2-1.2-3.1-1.2-4.2,0l-23.7,23.9L7,94.7c-1.7-1.6-4.5-0.5-4.5,1.9V142c0,1.8,1.5,3.3,3.3,3.3   h46.1c2.3,0,3.5-2.8,1.8-4.5l-14.5-14.3l24.1-23.1c1.2-1.2,1.2-3.1,0-4.3L50.1,85.8z"></path><path d="M143,95.7l-15.2,15l-24.4-24.6c-1.2-1.2-3.1-1.2-4.2,0L85.8,99.5c-1.2,1.2-1.2,3.1,0,4.3l24.9,23.8l-14.5,14.3   c-1.7,1.6-0.5,4.5,1.8,4.5h46.1c1.8,0,3.3-1.5,3.3-3.3V97.6C147.5,95.2,144.7,94.1,143,95.7z"></path><path d="M97.9,71.9L77.1,51c-1.3-1.3-3.3-1.3-4.6,0L51.6,71.9c-1.3,1.3-1.3,3.3,0,4.6l20.9,20.9c1.3,1.3,3.3,1.3,4.6,0   l20.9-20.9C99.2,75.2,99.2,73.1,97.9,71.9z"></path></g></svg>`
```
Comment (Haroenv) [inline js/inspiring-online.js]: good that you used svg

File: js/inspiring-online.js [hunk 2]
```diff
@@ -182,6 +182,14 @@ function renderPost(postData) {
     content.innerHTML = postData.content;
     element.appendChild(content);
 
+    var postUrl = '/' + postData.title.replace(/\s+/g, '-').replace(/[^a-z0-9+][^\w.-]/gi, '').toLowerCase() + '/'
```
Comment (Haroenv) [inline js/inspiring-online.js]: Do we have a wiki about post titles? This needs to be consistent for all of them.

Comment (TJScalzo) [inline js/inspiring-online.js]: Yes, I added it to the [Basic Post Structure](https://github.com/tholman/inspiring-online/wiki/Basic-Post-Structure) page. Should it be its own page?

**Comment by TJScalzo** [comment]
Which version of the icon is better?
screen shot 2016-09-28 at 6 59 47 pm
or
screen shot 2016-09-28 at 8 37 46 pm

File: js/inspiring-online.js (hunk 1 above)
Comment (tholman) [inline js/inspiring-online.js]: Lets keep the CSS in the CSS files though, not inline in the JS

**Comment by tholman** [comment]
Hmmm, I'm still kinda weird about this. I think people share the links to the actual content, rather than to a post page a lot more often... which was kind of one of the big ideas behind the redesign/reshuffle

File: js/inspiring-online.js (hunk 1 above)
Comment (TJScalzo) [inline js/inspiring-online.js]: Ok, I'll move it to the stylesheet when I get to my computer.

**Comment by TJScalzo** [comment]
@tholman My main reason for making this was to make it possible for people to share the fact that they got the website from this site. This feature is a way to share a post's content while also sharing the fact that they found it through inspiring.online. It's a way to get a post's page url if you need it.
I decided to try and make this after wanting to share a post using the method described above and I realized there's no easy way to find that link.

Would it be better if the maximize button was smaller/less intrusive?

**Comment by vaidaslamanauskas** [comment]
I'm more on the fence with Tim. The whole redesign goal was to make it straight forward, less clutter, less text if you like. With this, it seems like we are slightly introducing it back.

**Comment by TJScalzo** [comment]
Yeah, this isn't meant to be a super feature-rich site. Its purpose is meant to be very focused on getting users to the content they're looking for. If a majority of people think this feature isn't needed and/or wouldn't benefit users enough then this shouldn't be merged.

**Comment by TJScalzo** [comment]
What if instead of a button, the title at the bottom of the tile expands to the post's page? It won't add anything visually to the tiles to clutter them up.

One of the problems with doing this is it might mess up the way people think the sites work. 
The only other problem is how that line of text is supposed to work according to @tholman in #228.
Either way, it's an idea. Let me know what you guys think.

**Comment by vaidaslamanauskas** [comment]
Hmm..But actually, after redesign, what is the point of expanding the post. All of the content is there already.

**Comment by Haroenv** [comment]
In case you want to share the post and also the content, let people know of inspiring.online while letting them know of the article

**Comment by TJScalzo** [comment]
I'm working on a much more discreet place to put the maximize button. We'll see if it turns out to be worth anything.

**Comment by TJScalzo** [comment]
In 56630e7 I made the maximize button appear when hovering over the image. It works surprisingly well on the first set of loaded posts:
screen shot 2016-10-14 at 9 34 08 am
But it doesn't work when on a post page or when a second set of the infinitely scrolling posts is loaded. 

While the effect is interesting, it doesn't work everywhere on the site. That, combined with the fact that a couple of people have expressed concerns that this doesn't fit in with the site, makes me feel unsure of if I should continue work. Please let me know if anyone thinks this feature warrants more work other than me.

**Comment by vaidaslamanauskas** [comment]
Personally, I think it just adds confusion. Person sees expand icon: What's next? Will the post become bigger? Will it reveal more information? Will it open in new tab? Well...No...Its just moves to the top 😅

**Comment by tholman** [comment]
Yeah, I appreciate it... but at the same time, inspiring online doesn't really add much more content to be viewed, its more about sending them to where they want to go. I'd not share these pages really.

**Comment by TJScalzo** [comment]
Yep, that's what I was thinking. It was fun challenging myself with this feature. I'll close this pull request and delete the branch.

**Pull request closed without merging**